pennylane/templates/subroutines/qft.py

```diff
@@ -214,11 +214,13 @@ def compute_qfunc_decomposition(*wires, n_wires):  # pylint: disable=arguments-d
         def outer_loop(i):
             qml.Hadamard(wires[i])
 
-            @qml.for_loop(shift_len - i)
-            def cphaseshift_loop(j):
-                qml.ControlledPhaseShift(shifts[j], wires=[wires[i + j + 1], wires[i]])
+            if shift_len != 0:
```

Comment:
I think for this to be dynamic, we should use `qml.cond`? maybe we can use something like this eat the end?


```suggestion
    def single_wire_func():
        Hadamard(wires[0])

    qml.cond(shift_len == 0, single_wire_func, outer_loop)()
```

Reply:
actually the jaxpr it produces for 1 or multi wires, with/without autograph seem to be the same. So `if` should be ok.